src/base-element.js

```diff
@@ -414,6 +481,26 @@ export class BaseElement {
     return false;
   }
 
+  /**
+   * Ensure that the element is being eagerly loaded.
+   *
+   * Only used for V1 elements.
+   */
+  ensureLoaded() {}
+
+  /**
+   * Update the current `readyState`.
+   *
+   * Only used for V1 elements.
+   *
+   * @param {!./ready-state.ReadyState} state
+   * @param {*=} opt_failure
+   * @final
+   */
+  setReadyState(state, opt_failure) {
```

Comment:
What are the expectations around this fn?

- Is it valid to call within a ctor? Or do you need to wait for buildCallback at least?
- Is it valid to call at any point after buildCallback?
- Is it valid for elements to ever call this fn. with `UPGRADING` or `BUILDING`?

Reply:
It can be called any time. But calling before `buildCallback` doesn't make much sense since the element is automatically set to "building" by then. Setting to "upgrading" and "building" is theoretically possible, but doesn't seem very useful. The overall rule, however: once an element sets it first time, it basically "takes over" the state and should set it each time.